rkt/cli_apps.go

```diff
@@ -150,6 +155,90 @@ func (ae *appExec) Set(s string) error {
 	return nil
 }
 
+// appInjectVolume is for --inject-volume flags in the form of: --inject-volume source=PATH,target=PATH
+type appInjectVolume apps.Apps
+
+func (aam *appInjectVolume) Set(s string) error {
+	app := (*apps.Apps)(aam).Last()
+	if app == nil {
+		return fmt.Errorf("--inject-volume must follow an image")
+	}
+
+	p := strings.SplitN(s, ":", 2)
+	if len(p) != 2 {
+		return fmt.Errorf("must be SOURCE_PATH:TARGET_PATH")
+	}
+
+	volName, err := types.SanitizeACName(p[1])
+	if err != nil {
+		return fmt.Errorf("must be SOURCE_PATH:TARGET_PATH")
+	}
+
+	// TODO(iaguis) make it a random name?
+	rVol := types.Volume{Name: *types.MustACName(volName), Source: p[0], Kind: "host"}
+	mount := schema.Mount{Volume: rVol.Name, Path: p[1]}
+
+	(*apps.Apps)(aam).Volumes = append((*apps.Apps)(aam).Volumes, rVol)
+	if (*apps.Apps)(aam).Count() == 0 {
+		(*apps.Apps)(aam).Mounts = append((*apps.Apps)(aam).Mounts, mount)
+	} else {
+		app := (*apps.Apps)(aam).Last()
+		app.Mounts = append(app.Mounts, mount)
+	}
+
+	return nil
+}
+
+func (aam *appInjectVolume) Type() string {
+	return "appInjectVolume"
+}
+
+func (aam *appInjectVolume) String() string {
+	return ""
+}
+
+// appMount is for --mount flags in the form of: --mount volume=VOLNAME,target=PATH
+type appMount apps.Apps
+
+func (al *appMount) Set(s string) error {
+	mount := schema.Mount{}
+
+	// this is intentionally made similar to types.VolumeFromString()
+	// TODO(iaguis) use MakeQueryString() when appc/spec#520 is merged
+	m, err := url.ParseQuery(strings.Replace(s, ",", "&", -1))
+	if err != nil {
+		return err
+	}
+
+	for key, val := range m {
+		if len(val) > 1 {
+			return fmt.Errorf("label %s with multiple values %q", key, val)
+		}
+		switch key {
+		// FIXME(vc): ACName seems a bit restrictive for naming volumes and mountpoints...
+		case "volume":
+			mv, err := types.NewACName(val[0])
+			if err != nil {
+				return err
+			}
+			mount.Volume = *mv
+		case "target":
+			mount.Path = val[0]
+		default:
+			return fmt.Errorf("unknown mount parameter %q", key)
+		}
+	}
+
+	if (*apps.Apps)(al).Count() == 0 {
+		(*apps.Apps)(al).Mounts = append((*apps.Apps)(al).Mounts, mount)
```

Comment:
Minor - maybe create a separate pointer variable before the if doing the cast and use it here and below?